File: 백준/Silver/10815.숫자 카드/숫자 카드.java
import java.util.*;
import java.io.*;

public class Main {
    static int[] a;
    static int[] b;

    static boolean BinarySearch(int target) {
        int start = 0;
        int end = a.length - 1;
        while (start <= end) {
            int mid = (start + end) / 2;
            if (a[mid] == target) {
                return true;
            } else if (a[mid] > target) {
                end = mid - 1;
            } else if (a[mid] < target) {
                start = mid + 1;
            }
        }
        return false;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        a = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            a[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(a);

        int m = Integer.parseInt(br.readLine());
        b = new int[m];
        st = new StringTokenizer(br.readLine());

        for (int i = 0; i < m; i++) {
            b[i] = Integer.parseInt(st.nextToken());
        }
        StringBuilder sb = new StringBuilder();
        for(int i = 0; i < b.length;i++){
            if(BinarySearch(b[i])){
                sb.append(1).append(" ");
            }else{
                sb.append(0).append(" ");
            }
        }
        System.out.println(sb);
    }
}

File: 백준/Silver/10815. 숫자 카드/숫자 카드.py
import sys

def bs(target):
    start = 0
    end = len(has) - 1

    while start <= end:
        mid = (start + end) // 2

        if has[mid] == target:
            return True
        elif has[mid] < target:
            start = mid + 1
        else:
            end = mid - 1

    return False



n = int(sys.stdin.readline())
has = list(map(int,sys.stdin.readline().split()))

has.sort()

m = int(sys.stdin.readline())

find = list(map(int,sys.stdin.readline().split()))

result = []
for i in range(m):
    if bs(find[i]):
        result.append(1)
    else:
        result.append(0)


print(*result)